gtsam/hybrid/HybridBayesTree.cpp

```diff
@@ -41,6 +42,16 @@ bool HybridBayesTree::equals(const This& other, double tol) const {
   return Base::equals(other, tol);
 }
 
+/* ************************************************************************* */
+DiscreteValues HybridBayesTree::discreteMaxProduct(
+    const DiscreteFactorGraph& dfg) const {
+  TableFactor product = TableProduct(dfg);
+
+  uint64_t maxIdx = TableDistribution(product).argmax();
```

Comment:
This leaks privileged information outside of TableDistribution. I think the uint64 argmax should be internal, and like argmax in other factors/conditionals, an argmax should return a DiscreteValues.

Reply:
I agree with the comment but FYI except for `DiscreteLookupDAG` no argmax returns a DiscreteValues. That was why I originally did it this way.